hardhat.config.ts

```diff
@@ -0,0 +1,168 @@
+// Native
+import { readFileSync } from "fs";
+
+// Vendor
+import Joi from "joi";
+import toml from "toml";
+import "@nomiclabs/hardhat-waffle";
+import "@nomiclabs/hardhat-ethers";
+import "@nomiclabs/hardhat-etherscan";
+import "hardhat-gas-reporter";
+import "solidity-coverage";
+import "hardhat-contract-sizer";
+import { HardhatUserConfig, subtask } from "hardhat/config";
+import { TASK_COMPILE_SOLIDITY_GET_SOURCE_PATHS } from "hardhat/builtin-tasks/task-names";
+
+// Utilities
+import { validateEnvConfig } from "./scripts/utilities/validateConfig";
+
+// Validate values from .env file
+const {
+  FORKING,
+  FORK_BLOCK,
+  SOLC_DEFAULT,
+  PRIVATE_KEY,
+  MNEMONIC,
+  ETH_RPC_URL,
+  ARBITRUM_RPC_URL,
+  CMC_API_KEY,
+  ETHERSCAN_API_KEY,
+} = validateEnvConfig(
+  Joi.object({
+    SOLC_DEFAULT: Joi.string().default("0.8.10"),
+    PRIVATE_KEY: Joi.string(),
+    MNEMONIC: Joi.string(),
+    FORK_BLOCK: Joi.number(),
+    FORKING: Joi.boolean().default(false),
+    ETH_RPC_URL: Joi.string().required(),
+    ARBITRUM_RPC_URL: Joi.string().default(""),
+    CMC_API_KEY: Joi.string().default(""),
+    ETHERSCAN_API_KEY: Joi.string(),
+  })
+);
+
+// Configure accounts
+const accounts = PRIVATE_KEY
+  ? [PRIVATE_KEY]
+  : {
+      mnemonic:
+        MNEMONIC ||
+        "test test test test test test test test test test test junk",
+    };
+
+// Inherit Foundry config
+let foundry: { default: { solc: string } };
+
+try {
+  const foundryFile = toml.parse(readFileSync("./foundry.toml", "utf-8"));
+
+  foundry = {
+    default: {
+      solc: foundryFile.default["solc-version"] || SOLC_DEFAULT,
+    },
+  };
+} catch (error) {
+  foundry = {
+    default: {
+      solc: SOLC_DEFAULT,
+    },
+  };
+}
+
+// Filter out .t.sol test files and regular Solidity files in the test directory
+subtask(TASK_COMPILE_SOLIDITY_GET_SOURCE_PATHS).setAction(
+  async (_, __, runSuper) => {
+    const paths = await runSuper();
+
+    return paths.filter(
+      (p: string) => !p.endsWith(".t.sol") && !p.includes("test")
+    );
+  }
+);
+
+const config: HardhatUserConfig & {
+  contractSizer: {
+    alphaSort: boolean;
+    disambiguatePaths: boolean;
+    runOnCompile: boolean;
+  };
+  etherscan: {
+    apiKey: string;
+  };
+} = {
+  paths: {
+    cache: "hh-cache",
+    sources: "./src",
+    tests: "./test",
+  },
+  contractSizer: {
+    alphaSort: false,
+    disambiguatePaths: false,
+    runOnCompile: true,
+  },
+  defaultNetwork: "hardhat",
+  networks: {
+    localhost: {},
+    hardhat: {
+      allowUnlimitedContractSize: false,
+      chainId: 1337,
+      forking: {
+        blockNumber: Number(FORK_BLOCK),
+        enabled: FORKING === true,
+        url: ETH_RPC_URL,
+      },
+    },
+    mainnet: {
+      url: ETH_RPC_URL,
+      accounts,
+    },
+    arbitrum: {
+      url: ARBITRUM_RPC_URL,
+      accounts,
```

Comment:
I've opted for a generalized `accounts`

If we know we always will use a private key to deploy we can simplify this